### PR TITLE
✨Add wildcard domain matching for linker

### PR DESCRIPTION
**Pull request by calebcordry**

### CODE REVIEW

File: extensions/amp-analytics/0.1/linker-manager.js [hunk 1]
```diff
@@ -311,6 +322,20 @@ export class LinkerManager {
     return true;
   }
 
+  /**
+   * Allows specified wildcard matching of domains.
+   * Example:
+   *    `*.foo.com` matches `amp.foo.com`
+   *    `*.foo.com*` matches `amp.foo.com.uk`
```
Comment (lannka) [inline extensions/amp-analytics/0.1/linker-manager.js]: We need to clarify the following cases, and add unit tests to cover them:
- does `*.foo.co*` match `me.foo.com`?
- does `*.foo.com` match `foo.com`?
- does `*.foo.com` match `a.b.foo.com`?

Needs GA's input on this, what do you expect? @zikas @avimehta

Comment (calebcordry) [reply extensions/amp-analytics/0.1/linker-manager.js]: As it is currently written:
> * does `*.foo.co*` match `me.foo.com`?
Yes
> * does `*.foo.com` match `foo.com`?
No (no dot)
> * does `*.foo.com` match `a.b.foo.com`?
Yes

Happy to change if not in line with expectations.

Comment (lannka) [reply extensions/amp-analytics/0.1/linker-manager.js]: That all sounds good to me. @avimehta @zikas to confirm.

Can we have a set of tests for this private method? Just to cover all the corner cases, and also express the expectations.

Comment (avimehta) [reply extensions/amp-analytics/0.1/linker-manager.js]: lgtm. I think it is supported but just to confirm:

`*.foo.co*` should match `me.foo.co.uk` as well.

Comment (calebcordry) [reply extensions/amp-analytics/0.1/linker-manager.js]: @avimehta yes that will match, added test case.

Also added unit tests specifically targeting this logic.

**Comment by calebcordry** [comment]
Yes, will make a follow up PR for docs.

**Comment by lannka** [comment]
@calebcordry please add documentation
